website/docs/advanced-concepts/version-control-with-git/README.mdx

```diff
@@ -0,0 +1,102 @@
+
+### Create a new repository
+
+Create a new repository on your git service and copy the `SSH` URL. The repository should be empty except `README.md`, `.gitignore`, and `LICENSE` files (These can be auto-created when you create a new repository.)
+
+### Generate a deploy key
+
+1. Open the app you want to connect to git and click on the `Connect GIT` button on the bottom bar.
+2. In the Git connection window, paste your repository’s `SSH` URL and click on `Generate key` to generate an `SSH` key.
+
+
+
+ <VideoEmbed host="youtube" videoId="oFfdIwhSPL8" title="Generating a Deploy key" caption="Generating a Deploy key"/>
+
+Appsmith supports two types of keys -  **ECDSA 256** and **RSA 4096**. Please pick the one that's compatible with your Git hosting service. 
+
+### Adding the deploy key in the Repository
+
+Copy the generated deploy key and add it to your repository. Please note that you have to **enable write access** for pushing the changes to the remote repository from Appsmith.
+
+<Tabs queryString="current-edition">
+<TabItem label="GitHub" value="github">
+
+* Open the settings of the repository and go to deploy keys.
+* Click **Add deploy key**. Add a title for the key and paste the generated key into the key section.
+* Check the **Allow write access** checkbox and add the key.
+
+  <VideoEmbed host="youtube" videoId="4xMHO4G2hEA" title="Adding the deploy key in the Repository | Github" caption="Adding the deploy key in the Repository | Github"/>
+
+</TabItem>
+<TabItem label="Gitlab" value="gitlab">
+
+* Open the Settings> Repository and expand the deploy keys section.
+* Click **Add key** Add the title and paste the generated key in the key section.
+* Check the **Grant write permissions**" to this key and add the key.
+
+ <VideoEmbed host="youtube" videoId="9aaiE6OERW0" title="Adding the deploy key in the Repository | Gitlab" caption="Adding the deploy key in the Repository | Gitlab"/>
+
+</TabItem>
+<TabItem label="Bitbucket" value="bitbucket">
+
+* Open **Personal settings** from the bottom right corner and go to **SSH keys**.
+* Click **Add key**, add the label and paste the generated key in the key section.
+
+ <VideoEmbed host="youtube" videoId="A8ZOvW1CVIk" title="Adding the deploy key in the Repository | Bitbucket" caption="Adding the deploy key in the Repository | Bitbucket"/>
+
+</TabItem>
+</Tabs>
+
+:::info
+**AWS CodeCommit** isn't supported as a source control repository due to their lack of support for deploy keys.
+:::
+
+### User configurations
+
+* In user settings, define the user configurations. By default, the global user configurations are used. Click the **Edit** button or directly open the profile section to change the global configurations. If you want to add a user configuration specific to the current app, you must clear the **use default configuration** checkbox and enter the author details below.
```

Comment:
there is some confusion here.  We are not communicating clearly to the user what global configuration means. They will try to find the Edit button in the git connect modal.  It would take them a while to figure that this on the workspace profile under Git user config.  Let's rewrite this para to communicate this clearly.